ann_benchmarks/main.py

```diff
@@ -118,14 +110,40 @@ def main():
         print('running only', args.algorithm)
         definitions = [d for d in definitions if d.algorithm == args.algorithm]
 
-    if args.docker_tag:
-        print('running only', args.docker_tag)
-        definitions = [d for d in definitions if d.docker_tag == args.docker_tag]
-
-    if set(d.docker_tag for d in definitions).difference(docker_tags):
-        print('not all docker images available, only:', set(docker_tags))
-        print('missing docker images:', set(d.docker_tag for d in definitions).difference(docker_tags))
-        definitions = [d for d in definitions if d.docker_tag in docker_tags]
+    if not args.local:
+        # See which Docker images we have available
+        docker_client = docker.from_env()
+        docker_tags = set()
+        for image in docker_client.images.list():
+            for tag in image.tags:
+                tag, _ = tag.split(':')
+                docker_tags.add(tag)
+
+        if args.docker_tag:
+            print('running only', args.docker_tag)
+            definitions = [d for d in definitions if d.docker_tag == args.docker_tag]
+
+        if set(d.docker_tag for d in definitions).difference(docker_tags):
+            print('not all docker images available, only:', set(docker_tags))
+            print('missing docker images:', set(d.docker_tag for d in definitions).difference(docker_tags))
+            definitions = [d for d in definitions if d.docker_tag in docker_tags]
+    else:
+        def _test(df):
```

Comment:
seems a bit unnecessary to have this function here, but that's a fairly minor point